src/types.ts

```diff
@@ -78,16 +78,25 @@ export interface StdSignature {
  * An unsigned transaction.
  */
 export interface Tx {
+    type: string;
+    value: {
     msg: Msg[];
     fee: StdFee;
     memo: string;
+    };
 }
 
 /**
  * A signed transaction.
  */
 export interface StdTx extends Tx {
+    type: string;
+    value: {
+    msg: Msg[];
+    fee: StdFee;
+    memo: string;
     signatures: StdSignature[];
+    };
```

Comment:
Pretty sure `extends` will automatically include attributes from `Tx`.

```suggestion
    signatures: StdSignature[];
```

Reply:
Because it's embedded in the 'value' object, typescript compiler was complaining. Related to this I believe: https://stackoverflow.com/questions/53636756/typescript-interface-extending-another-interface-with-nested-properties

Reply:
Ah, okay. In that case, the duplication here is fine - no need for intersection types here.